app/src/lib/markdown-filters/video-link-filter.ts

```diff
@@ -0,0 +1,82 @@
+import { INodeFilter } from './node-filter'
+import { githubAssetVideoRegex } from './video-url-regex'
+
+/**
+ * The Video Link filter matches a github-flavored markdown target of a link,
+ * like
+ * <p><a href="https://user-images.githubusercontent.com/7559041/1234.mp4">…</a></p>.
+ *
+ * This type of pattern is formed when a user pastes the video url in markdown
+ * editor and then the markdown parser auto links it.
+ *
+ * If the url is in the format of a github user asset, it will replace the
+ * paragraph with link with a a video tag. If not, the link is left unmodified.
+ */
+export class VideoLinkFilter implements INodeFilter {
+  /**
+   * Video link matches on p tags with an a tag with a single child of a tag
+   * with an href that's host matches a pattern of video url that is a github
+   * user asset.
+   */
+  public createFilterTreeWalker(doc: Document): TreeWalker {
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    const filter = this
+    return doc.createTreeWalker(doc, NodeFilter.SHOW_ELEMENT, {
+      acceptNode: function (el: Element) {
+        return filter.getGithubVideoLink(el) === null
+          ? NodeFilter.FILTER_SKIP
+          : NodeFilter.FILTER_ACCEPT
+      },
+    })
```

Comment:
I'd prefer if we used a bound method here instead of disabling the eslint rule. We try to avoid capturing `this` if possible

```suggestion
    return doc.createTreeWalker(doc, NodeFilter.SHOW_ELEMENT, {
      acceptNode: (el: Element) =>
        this.getGithubVideoLink(el) === null
          ? NodeFilter.FILTER_SKIP
          : NodeFilter.FILTER_ACCEPT,
    })
```